File: go/152/main.go
package main

import (
	"fmt"
)

func allElementsEqual(nums []int) bool {
    if len(nums) == 0 {
        return true // 空配列は全て等しいとみなす
    }

    first := nums[0]
    for _, num := range nums {
        if num != first {
            return false
        }
    }
    return true
}

func main() {
    fmt.Println(allElementsEqual([]int{3, 3, 3, 3})) // true
    fmt.Println(allElementsEqual([]int{1, 2, 1}))    // false
    fmt.Println(allElementsEqual([]int{7}))          // true
}

File: python/152/main.py
def max_len_zero_sum_subarray(nums):
    prefix_sum = 0
    seen = {0: -1}
    max_len = 0

    for i, num in enumerate(nums):
        prefix_sum += num
        if prefix_sum in seen:
            max_len = max(max_len, i - seen[prefix_sum])
        else:
            seen[prefix_sum] = i

    return max_len


# テスト
print(max_len_zero_sum_subarray([1, -1, 5, -2, 3]))  # 出力: 4
print(max_len_zero_sum_subarray([-2, 1, 1, -1, 2]))  # 出力: 4
print(max_len_zero_sum_subarray([1, 2, 3]))  # 出力: 0
